types/mempool/sender_nonce.go

```diff
@@ -65,6 +87,12 @@ func (snm *senderNonceMempool) setSeed(seed int64) {
 // Insert adds a tx to the mempool. It returns an error if the tx does not have at least one signer.
 // priority is ignored.
 func (snm *senderNonceMempool) Insert(_ sdk.Context, tx sdk.Tx) error {
+	if snm.maxTx > 0 && len(snm.existingTx) >= snm.maxTx {
+		return ErrMempoolTxMaxCapacity
+	}
+	if snm.maxTx < 0 {
+		return nil
+	}
```

Comment:
@odeke-em, what do you think of adding this extra condition, the result being negative means unbounded, positive bounded, and 0 disabled? Given that the main request was the ability to turn it off. also, what do you guys think @tac0turtle @kocubinski @alexanderbez

Reply:
Given the three behaviors we want this makes sense to me.
| maxTx | behavior |
| --- | --|
| negative | disabled (insert returns early) |
| zero | unbounded (insert never fails with ErrMempoolTxMaxCapacity |
| positive | bounded (insert may fail with ErrMempoolTxMaxCapacity) |

Reply:
@JeancarloBarrios This behavior needs to be documented too.

Reply:
Sorry, that is what I meant thanks for the correction.